borg/helpers.py

```diff
@@ -1107,9 +1114,17 @@ class FakeArchive:
             b'mode': 0, b'path': '', b'user': '', b'group': '', b'mtime': 0,
             b'uid': 0, b'gid': 0,
         }).keys())
-        keys.sort()
-        keys.sort(key=str.isupper)
-        return keys
+        return sorted(sorted(keys), key=str.isupper)
```

Comment:
first alphabetical sort, then separate keys only composed of uppercase letters to the bottom. This way the "formatting" aids (TAB, NEWLINE, LF, CR, NUL) appear together at the bottom, not mixed between content keys.